### PR TITLE
Minor enhancements to ci_folder_parsing script

### PR DESCRIPTION
**Pull request by lubnarasheed**
## Description
Updated ci_parsing_folder script to accept argument '--number' or '-n' to specify which test output to run

## Motivation and Context
Updated ci_parsing_folder script to accept argument '--number' or '-n' to specify which test output to run

## Impact (If any)
N/A

## Screenshots:
image

```
(py381) LRASHEED-M-43JH:tests lrasheed$ python ci_folder_parsing.py -o iosxe -c ShowPolicyMapInterface -n 8
2020-10-08T10:40:06: %AETEST-INFO: +------------------------------------------------------------------------------+
2020-10-08T10:40:06: %AETEST-INFO: |                       Starting testcase FileBasedTest                        |
2020-10-08T10:40:06: %AETEST-INFO: +------------------------------------------------------------------------------+
2020-10-08T10:40:06: %AETEST-INFO: +------------------------------------------------------------------------------+
2020-10-08T10:40:06: %AETEST-INFO: |           Starting section check_os_folder[operating_system=iosxe]           |
2020-10-08T10:40:06: %AETEST-INFO: +------------------------------------------------------------------------------+
2020-10-08T10:40:06: %AETEST-INFO: +..............................................................................+
2020-10-08T10:40:06: %AETEST-INFO: :               Starting STEP 1: iosxe -> ShowPolicyMapInterface               :
2020-10-08T10:40:06: %AETEST-INFO: +..............................................................................+
2020-10-08T10:40:06: %AETEST-INFO: +..............................................................................+
2020-10-08T10:40:06: %AETEST-INFO: :      Starting STEP 1.1: Test Golden -> iosxe -> ShowPolicyMapInterface       :
2020-10-08T10:40:06: %AETEST-INFO: +..............................................................................+
2020-10-08T10:40:06: %AETEST-INFO: +..............................................................................+
2020-10-08T10:40:06: %AETEST-INFO: :    Starting STEP 1.1.1: ('Gold -> iosxe -> ShowPolicyMapInterface -> gold    :
2020-10-08T10:40:06: %AETEST-INFO: :                                en_output8',)                                 :
2020-10-08T10:40:06: %AETEST-INFO: +..............................................................................+
2020-10-08T10:40:06: %AETEST-INFO: The result of STEP 1.1.1: ('Gold -> iosxe -> ShowPolicyMapInterface -> golden_output8',) is => PASSED
2020-10-08T10:40:06: %AETEST-INFO: The result of STEP 1.1: Test Golden -> iosxe -> ShowPolicyMapInterface is => PASSED
2020-10-08T10:40:06: %AETEST-INFO: +..............................................................................+
2020-10-08T10:40:06: %AETEST-INFO: :       Starting STEP 1.2: Test Empty -> iosxe -> ShowPolicyMapInterface       :
2020-10-08T10:40:06: %AETEST-INFO: +..............................................................................+
2020-10-08T10:40:06: %AETEST-INFO: +..............................................................................+
2020-10-08T10:40:06: %AETEST-INFO: :    Starting STEP 1.2.1: Empty -> iosxe -> ShowPolicyMapInterface -> empty    :
2020-10-08T10:40:06: %AETEST-INFO: :                                   _output                                    :
2020-10-08T10:40:06: %AETEST-INFO: +..............................................................................+
2020-10-08T10:40:06: %AETEST-INFO: The result of STEP 1.2.1: Empty -> iosxe -> ShowPolicyMapInterface -> empty_output is => PASSED
2020-10-08T10:40:06: %AETEST-INFO: The result of STEP 1.2: Test Empty -> iosxe -> ShowPolicyMapInterface is => PASSED
2020-10-08T10:40:06: %AETEST-INFO: The result of STEP 1: iosxe -> ShowPolicyMapInterface is => PASSED
2020-10-08T10:40:06: %AETEST-INFO: +----------------------------------------------------------+
2020-10-08T10:40:06: %AETEST-INFO: |                       STEPS Report                       |
2020-10-08T10:40:06: %AETEST-INFO: +----------------------------------------------------------+
2020-10-08T10:40:06: %AETEST-INFO: STEP 1 - iosxe -> ShowPolicyMapInterface              Passed
2020-10-08T10:40:06: %AETEST-INFO: STEP 1.1 - Test Golden -> iosxe -> ShowPolicyMapInterface    Passed
2020-10-08T10:40:06: %AETEST-INFO: STEP 1.1.1 - ('Gold -> iosxe -> ShowPolicyMapInterface -> golden_output8',)    Passed
2020-10-08T10:40:06: %AETEST-INFO: STEP 1.2 - Test Empty -> iosxe -> ShowPolicyMapInterface    Passed
2020-10-08T10:40:06: %AETEST-INFO: STEP 1.2.1 - Empty -> iosxe -> ShowPolicyMapInterface -> empty_output    Passed
2020-10-08T10:40:06: %AETEST-INFO: ------------------------------------------------------------
2020-10-08T10:40:06: %AETEST-INFO: The result of section check_os_folder[operating_system=iosxe] is => PASSED
2020-10-08T10:40:06: %AETEST-INFO: The result of testcase FileBasedTest is => PASSED
2020-10-08T10:40:06: %AETEST-INFO: +------------------------------------------------------------------------------+
2020-10-08T10:40:06: %AETEST-INFO: |                               Detailed Results                               |
2020-10-08T10:40:06: %AETEST-INFO: +------------------------------------------------------------------------------+
2020-10-08T10:40:06: %AETEST-INFO:  SECTIONS/TESTCASES                                                      RESULT
2020-10-08T10:40:06: %AETEST-INFO: --------------------------------------------------------------------------------
2020-10-08T10:40:06: %AETEST-INFO: .
2020-10-08T10:40:06: %AETEST-INFO: `-- FileBasedTest                                                         PASSED
2020-10-08T10:40:06: %AETEST-INFO:     `-- check_os_folder[operating_system=iosxe]                           PASSED
2020-10-08T10:40:06: %AETEST-INFO:         |-- Step 1: iosxe -> ShowPolicyMapInterface                       PASSED
2020-10-08T10:40:06: %AETEST-INFO:         |-- Step 1.1: Test Golden -> iosxe -> ShowPolicyMapInterface      PASSED
2020-10-08T10:40:06: %AETEST-INFO:         |-- Step 1.1.1: ('Gold -> iosxe -> ShowPolicyMapInterface -...    PASSED
2020-10-08T10:40:06: %AETEST-INFO:         |-- Step 1.2: Test Empty -> iosxe -> ShowPolicyMapInterface       PASSED
2020-10-08T10:40:06: %AETEST-INFO:         `-- Step 1.2.1: Empty -> iosxe -> ShowPolicyMapInterface ->...    PASSED
2020-10-08T10:40:06: %AETEST-INFO: +------------------------------------------------------------------------------+
2020-10-08T10:40:06: %AETEST-INFO: |                                   Summary                                    |
2020-10-08T10:40:06: %AETEST-INFO: +------------------------------------------------------------------------------+
2020-10-08T10:40:06: %AETEST-INFO:  Number of ABORTED                                                            0
2020-10-08T10:40:06: %AETEST-INFO:  Number of BLOCKED                                                            0
2020-10-08T10:40:06: %AETEST-INFO:  Number of ERRORED                                                            0
2020-10-08T10:40:06: %AETEST-INFO:  Number of FAILED                                                             0
2020-10-08T10:40:06: %AETEST-INFO:  Number of PASSED                                                             1
2020-10-08T10:40:06: %AETEST-INFO:  Number of PASSX                                                              0
2020-10-08T10:40:06: %AETEST-INFO:  Number of SKIPPED                                                            0
2020-10-08T10:40:06: %AETEST-INFO:  Total Number                                                                 1
2020-10-08T10:40:06: %AETEST-INFO:  Success Rate                                                            100.0%
2020-10-08T10:40:06: %AETEST-INFO: --------------------------------------------------------------------------------
(py381) LRASHEED-M-43JH:tests lrasheed$
```

## Checklist:

- [x] I have updated the changelog.
- [x] I have updated the documentation (If applicable).
- [x] I have added tests to cover my changes (If applicable).
- [x] All new and existing tests passed.
- [x] All new code passed compilation.

### CODE REVIEW

**Comment by lubnarasheed** [comment]
@itdependsnetworks would you be able to review this as well?

File: tests/ci_folder_parsing.py [hunk 1]
```diff
@@ -29,11 +34,16 @@
...
+my_parser.add_argument('-n', "--number",
+                       type=str,
+                       help="The specific unittest we want to run, such as '25'",
+                       default=None)
 args = my_parser.parse_args()
 
 _os = args.operating_system
 _class = args.class_name
 _token = args.token
+_number = args.number
```
Comment (itdependsnetworks) [inline tests/ci_folder_parsing.py:46]: Does it make sense to ensure if _number is assigned, that so us _os and _class?

**Comment by itdependsnetworks** [comment]
Made one comment, but wouldn't see it as a blocker

File: tests/ci_folder_parsing.py (hunk 1 above)
Comment (lubnarasheed) [reply tests/ci_folder_parsing.py:47]: * for _n, I've changed the type to be an integer to ensure something like ``10.5`` is not passed through
* for _os, I've added a check to ensure the argument is provided
* for _class, I've added a check to provide an error if -n is provided without -c

**Review by tahigash — APPROVED** [review]

File: tests/ci_folder_parsing.py [hunk 2]
```diff
@@ -46,8 +46,8 @@
 _token = args.token
 _number = args.number
 
-if _number and not _class or not _os:
-    sys.exit("Unittest number provided without specifying supporting arguments:"
+if _number and not _class or _number and not _os:
```
Comment (jeaubin) [inline tests/ci_folder_parsing.py:49]: or _number and not _os:

i dont understand this use case, can you explain it to me?  It should be both, class and os  no?

**Review by jeaubin — APPROVED** [review]

**Pull request merged**